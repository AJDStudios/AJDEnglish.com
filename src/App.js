import React, { useState } from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Header from './components/Header.js';
import Footer from "./components/Footer";
import Services from "./pages/Services.js";
import AboutAD from "./pages/AboutAD.js";
import Testimonials from "./pages/Testimonials.js";
import MyMission from "./pages/MyMission.js";
import Contact from "./pages/contact.jsx";
import SplashScreen from './pages/SplashScreen.js';
import './styles/App.css';

const App = () => {
  const [isLoading, setIsLoading] = useState(true); 

  setTimeout(() => {
    setIsLoading(false);
  }, 3000);

  return (
    <div className="app">
      {isLoading ? (
        <SplashScreen />
      ) : (
        <Router>
          <Header />
            <Routes>
              <Route path="/" element={<SplashScreen />} />
              <Route path="/Services" element={<Services />} /> 
              <Route path="/AboutAD" element={<AboutAD />} />
              <Route path="/Testimonials" element={<Testimonials />} />
              <Route path="/MyMission" element={<MyMission />} />
              <Route path="/Contact" element={<Contact />} />
            </Routes>
          <Footer /> 
        </Router>
      )}
    </div>
  );
};

export default App;
