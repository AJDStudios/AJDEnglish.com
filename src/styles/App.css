* {
  margin: 0;
  padding: 0;
  box-sizing: border-box; 
}

.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: space-between; 
  min-height: 100vh;
  overflow-x: hidden;
}

body {
  margin: 0;
  padding: 0;
  background-color: beige;
  font-family: Helvetica, sans-serif;
  font-weight: 300;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
  padding: 0;
  margin: 0;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: beige;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
