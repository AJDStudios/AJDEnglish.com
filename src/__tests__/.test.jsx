import React from 'react';
import { render, screen, act } from '@testing-library/react';
import App from '../App';
import '@testing-library/jest-dom/extend-expect';

// Renders the app and verifies snapshot
test('renders the app component', () => {
  const { asFragment } = render(
    <App />
  );
  expect(asFragment()).toMatchSnapshot();
});

// Checks that no redundant code or comments exist
test('does not contain redundant code or comments', () => {
  const { queryByText } = render(
    <App />
  );
  expect(queryByText('<!-- Notice the use of %PUBLIC_URL% in the tags above.')).toBeNull();
});

// Verifies that the splash screen is rendered correctly
test('renders the splash screen', () => {
  render(
    <App />
  );
  const splashScreen = screen.getByTestId('splash-screen');
  expect(splashScreen).toBeInTheDocument();
});

// Simulate initial loading state
describe('App Loading State', () => {
  test('renders with initial loading state', () => {
    act(() => {
      render(
        
        <App />

      );
    });
    const loadingIndicator = screen.getByTestId('splash-screen');
    expect(loadingIndicator).toBeInTheDocument();
  });

  test('renders loading state when isLoading is true', () => {
    act(() => {
      render(

        <App isLoading={true} />

      );
    });
    const loadingIndicator = screen.getByTestId('splash-screen');
    expect(loadingIndicator).toBeInTheDocument();
  });
});
