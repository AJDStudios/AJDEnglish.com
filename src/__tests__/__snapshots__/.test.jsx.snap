// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app component 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="splash-screen"
      data-testid="splash-screen"
    >
      <div
        class="splash-header"
      >
        <img
          alt="Selfie of AD with students"
          class="splash-image"
          src="test-file-stub"
        />
      </div>
      <div
        class="splash-text"
      >
        <p>
          Welcome to A J D English!
          <br />
          <br />
          Experience flexible, student-led lessons designed to build your confidence and communication skills.
          <br />
          <br />
          Forget the traditional classroomâ€”our online sessions are tailored to your needs, allowing you to learn at your pace, on your schedule.
          <br />
          <br />
          Speak fluently and comfortably in any situation.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
